### DISCUSSION TITLE
Focusing on Vehicle Physics in the new Godot Physics Enginr

### DISCUSSION THREAD

**Discussion by MonochromeSilence8807** [godotengine/godot-proposals#6336] [3D]
Hello. I would like to see some focus of the devs on the vehicle physics in Godot, so them can be more realistic. In the future.

Thanks.

**Comment by Zireael07** [comment]
There is talk about moving the vehicle node to an addon, so I wouldn't expect that.

PS. Vehicle physics are a port of Bullet vehicle physics, which are widely used and believed to be realistic.

**Comment by MonochromeSilence8807** [comment]
Well then if it's going to move to an addon, at least make it as realistic as Bullet, cause there are people that need this. In Godot, yes.

**Reply by Zireael07** [reply]
It is literally "as realistic" as Bullet since it's literally a copy of Bullet's implementation

**Reply by MonochromeSilence8807** [reply]
yeah but since it doesn't have vehicle physics that's a big minus

**Reply by Zireael07** [reply]
What are you talking about? The VehicleBody3D and VehicleWheel3D are literally your vehicle physics

**Reply by MonochromeSilence8807** [reply]
Yeah but these things are not enough in my opinion. Maybe I just don't know anything, as I am a noob. How are the vehicle physics in Unreal? They have the same  VehicleBody and VehicleWheel options?

**Reply by Zireael07** [reply]
I am not sure about the code behind it, but functionally Unreal 4's vehicles are very similar to Bullet, yes

**Reply by MonochromeSilence8807** [reply]
Oh so I am just stupid, thank you for clarifying that for me.

**Reply by Zireael07** [reply]
Tbh Bullet physics CAN be a bit hard to set up, you're best off following some sort of a tutorial, e.g. BastiaanOlij's vehicle demo...

**Reply by MonochromeSilence8807** [reply]
Yeah I've checked that video. Thank you.

**Comment by MonochromeSilence8807** [comment]
https://youtu.be/q6GdUQngOnA

Btw is that possible in Godot?

**Comment by BastiaanOlij** [comment]
The vehicle physics in Bullet and Godot are based on a raycast setup that has been implemented in many game engines and some very popular racing games. It's a nice approximation of vehicle physics that does well enough in many applications and just requires some work on the developers part to get familiar with and use well. 

But if you are after a realistic physics simulation like you find in games like Forza, Gran Turismo, Assetto Corsa and the like, this requires something far more specialized that has a complexity that falls way outside of the realm of what we could, or would want to, support in the core of Godot.

There are several developers working on plugins for various different implementations, but it is up to them whether they will make these available to the general public or not.

That all said, if you are new to this my best advise to you is, start with Godots build in vehicle setup as it is right now, and build something within its constraints.
